packages/snaps-sdk/src/provider.ts

```diff
@@ -0,0 +1,18 @@
+import type { BaseProvider } from '@metamask/providers';
+
+type BaseProviderInstance = InstanceType<typeof BaseProvider>;
+
+/**
+ * The `window.ethereum` provider for Snaps.
+ */
+export type SnapsEthereumProvider = Pick<
+  BaseProviderInstance,
+  // Snaps have access to a limited set of methods. This is enforced by the
+  // `proxyStreamProvider` function in `@metamask/snaps-execution-environments`.
+  'request' | 'on' | 'removeListener'
+>;
+
+/**
+ * The `window.snap` provider.
+ */
+export { SnapsGlobalObject as SnapsProvider } from '@metamask/snaps-rpc-methods';
```

Comment:
Should we detach this from `snaps-rpc-methods`?

Reply:
The request type is inferred from `snaps-rpc-methods`. We could hardcode all the types, but then we'd have to make sure to update it.

Reply:
I wonder if we could set it up in a way that we only use snaps-sdk types in the rpc methods and therefore enforce having to update the types